lib/pure/asyncdispatch.nim

```diff
@@ -355,27 +355,50 @@ proc `or`*[T, Y](fut1: Future[T], fut2: Future[Y]): Future[void] =
   fut2.callback = cb
   return retFuture
 
-proc all*[A](futs: seq[Future[A]]): Future[seq[A]] =
+proc all*[T](futs: varargs[Future[T]]): auto =
```

Comment:
I wanted really badly to specify the type explicitly, but couldn't find a way to. Depending on the type of the input Futures, the return type is either Future[void] of Future[seq[T]]. I tried `Future[seq[T]] | Future[void]`, `Future[seq[T]] or Future[void]`, with no success. So I had to make it auto to make it work.

`void` handling added complexity and dirtiness to the code, but I really can't see a better way to do it. Suggestions are very much welcome.